corehq/apps/users/models.py

```diff
@@ -2096,6 +2096,11 @@ def __repr__(self):
             self=self
         ))
 
+    @property
+    @memoized
```

Comment:
nit: I think we can start using cached_property https://docs.djangoproject.com/en/1.11/ref/utils/#django.utils.functional.cached_property